data-protocols/ids/ids-core/src/main/java/org/eclipse/dataspaceconnector/ids/core/service/CatalogServiceImpl.java

```diff
@@ -49,10 +52,11 @@ public CatalogServiceImpl(
      */
     @Override
     @NotNull
-    public Catalog getDataCatalog(ClaimToken claimToken, Range range) {
-        var query = ContractOfferQuery.Builder.newInstance().claimToken(claimToken).build();
+    public Catalog getDataCatalog(ClaimToken claimToken, Range range, List<Criterion> filters) {
 
-        var offers = contractOfferService.queryContractOffers(query, range).collect(toList());
+        var query = ContractOfferQuery.Builder.newInstance().claimToken(claimToken).assetsCriteria(filters).definitionsRange(range).build();
```

Comment:
please just call it `range` instead of `definitionsRange` - we have another issue where the semantics of it will change, and the name would be misleading then.

Reply:
@paullatzelsperger I suggested this name, because right now it's a range on the definitions, and it's good to have it explicit, it will avoid misunderstandings, renaming it to `range` would be something for the bugfix.

Reply:
@paullatzelsperger @ndr-brt so what's the conclusion here? Should we leave it as it is (definitionsRange) for now, and change it later?